### PR TITLE
feat(semantic-release): set up automated releases

### PR DESCRIPTION
**Pull request by elizabethsjudd**
### Summary

 Sets up automated releases using the semantic release package. Have to use the `feat` commit message format to trigger an actual release (everything that's been done since the last release has been non-release types.  Using the default semantic release configurations. If we want to change them, we can update to use IBM/test-automation tools for commit messages.

Note: I'll need to manually update the changelog notes to mention the TypeScript work that was done prior to this.

### Added

- Adds the semantic release package

### Updated

- Travis configurations to run semantic release
- Broke up Travis jobs for naming

## Code review

- [x] The code must be reviewed by at least one Maintainer

## Definition of done

- The code has been reviewed by the Maintainer
- The code has passed quality checks
  - 85% code coverage
  - Passed the following tests:
    - Unit
- Documents are updated

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,12 +56,22 @@
     "mocha": "7.1.2",
     "rollup": "2.9.1",
     "rollup-plugin-terser": "5.3.0",
+    "semantic-release": "^17.0.8",
```
Comment (scottnath) [inline package.json]:
```suggestion
    "semantic-release": "17.0.8",
```

File: package.json [hunk 2]
```diff
@@ -56,12 +56,22 @@
...
-  }
+  },
+  "release": {
+		"branches": [
+			{
+				"name": "stable",
+				"channel": "latest"
+			}
+		],
+		"tagFormat": "v${version}"
```
Comment (scottnath) [inline package.json]:
```suggestion
		]
```
this is the [default](https://semantic-release.gitbook.io/semantic-release/usage/configuration#tagformat) so should automatically be there. Semrel is finicky though, so no worries if you ignore this :)

**Review by scottnath — APPROVED** [review]

**Review by cal-smith — APPROVED** [review]
Cool!

**Pull request merged**